test/OpenTelemetry.Tests/Metrics/AggregatorTest.cs

```diff
@@ -188,6 +188,26 @@ public void HistogramWithOnlySumCount()
             Assert.False(enumerator.MoveNext());
         }
 
+        internal static void AssertExponentialBucketsAreCorrect(Base2ExponentialBucketHistogram expectedHistogram, ExponentialHistogramData data)
+        {
+            Assert.Equal(expectedHistogram.Scale, data.Scale);
+            Assert.Equal(expectedHistogram.ZeroCount, data.ZeroCount);
+            Assert.Equal(expectedHistogram.PositiveBuckets.Offset, data.PositiveBuckets.Offset);
+            Assert.Equal(expectedHistogram.NegativeBuckets.Offset, data.NegativeBuckets.Offset);
+
+            var index = expectedHistogram.PositiveBuckets.Offset;
+            foreach (var bucketCount in data.PositiveBuckets)
```

Comment:
I think this should be tested by enumerating over the expected data instead of the actual data. That would ensure that each of the expected bucket counts were present in the actual data as well.

If `foreach (var bucketCount in data.PositiveBuckets)` does not return any item or returns lesser items than expected, we would not be able to verify that all the expected buckets were actually present.

Reply:
Yup, this would have caught the missed copy of size.